### DISCUSSION TITLE
Using subscription as trigger function

### DISCUSSION THREAD

**Discussion by phappy** [FreeOpcUa/opcua-asyncio#487] [Q&A]
Hi everybody,

I'm quite new to OPC UA and Python. I have a question about NodeIDs in a subscription.
I got an OPC Sever with several nodes. I want to get the data change of one node by using subscription and use this a a trigger as it seems to be the fastest way.

When the value in my subscribtion is changing, I want to read other NodeIDs and store them. I think I need to extend the datachange_notification function in my Subhandler but I don't know how to access the other nodes in the handler or to get some return value

Thanks a lot

      def datachange_notification(self, node, val, data):
          #print("Python: New data change event", node, val)

          var_1_data = var_1.get_value()
          var_2_data = var_2.get_value()
          var_3_data = var_3.get_value()
  
          write_to_sql(var_1_data,var_2_data,var_3_data)
         
  
      try:
  
          client.connect()
          root = client.get_root_node()
  
          trigger = root.get_child([""])
          var_1 = client.get_node("ns=2;s=XXX")
          var_2 = client.get_node("ns=2;s=YYY")
          var_3 = client.get_node("ns=2;s=ZZZ")
  
          handler = SubHandler()
          sub = client.create_subscription(500, handler)
          handle = sub1.subscribe_data_change(trigger)

**Comment by AndreasHeine** [comment]
first of all this is not async code! this is **opcua-asyncio**

**Reply by AndreasHeine** [reply]
the datachange_notification() is alway called if a value changes and the docstrings in the sourcecode also tells you that you should not make long running requests inside of that callback!!! usually in the callback you just put it on a queue of your choise and never block the callback by polling data (which can take a lot of time) and then make a db request (which can take a lot of time) aswell.
